packages/typespec-python/test/mock_api_tests/test_apitest.py

```diff
@@ -0,0 +1,18 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+from apitest.limit import LimitClient
+from azure.core.exceptions import HttpResponseError
+
+@pytest.fixture
+def client():
+    with LimitClient() as client:
+        yield client
+
+# actually SDK does not validate the length of the name
+def test_limit_length(client: LimitClient):
+    with pytest.raises(HttpResponseError):
+        client.length(name="ok")
```

Comment:
Actually, client validation doesn't work in DPG because [validation](https://github.com/Azure/msrest-for-python/blob/af41991f17445e4cb26d39c247f091ddfd7027b9/msrest/serialization.py#L231-L255) in serialization is lacked compared with `msrest`

Reply:
hmm so we're not actually doing any path parameter validation right now in DPG, even if we generate with these parameters passed to the serialization function. I'm now starting a discussion about what we want to do for path parameter validation from tsp

Reply:
after talking with the DPG team, we don't want to add any information from the validation decorators to our generated code. At most what we would want to verify for path parameters is that they're not null / empty. As such, let me close this PR. Thanks for your work @msyyc, sorry we didn't end up using it, but by you showing that we aren't actually doing any information with the validation decorators in serialization for currently DPG, it started a bigger discussion about whether we want to even generate info from the validation decorators in tsp. Going to close this PR, thanks!

cc @tadelesh